bank-service/src/main/java/rs/banka4/bank_service/utils/ListingInfoScheduler.java

```diff
@@ -0,0 +1,213 @@
+import rs.banka4.bank_service.domain.security.Security;
+import rs.banka4.bank_service.repositories.ListingDailyPriceInfoRepository;
+import rs.banka4.bank_service.repositories.ListingRepository;
+import rs.banka4.bank_service.repositories.SecurityRepository;
+import rs.banka4.bank_service.runners.ListingsDataRunner;
+
+@Profile("!test")
+@Component
+@RequiredArgsConstructor
+public class ListingInfoScheduler {
+
+    private final SecurityRepository securityRepository;
+
+    private final ListingRepository listingRepository;
+
+    private final ListingDailyPriceInfoRepository listingDailyPriceInfoRepository;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ListingInfoScheduler.class);
+
+    // @Scheduled(fixedDelayString = "#{7l * 60l * 1000l}")
+    @Scheduled(
+        cron = "0 1 0 * * *",
+        zone = "Europe/Belgrade"
+    )
+    @Transactional
+    public void scheduleListingInfoUpdates() {
+        if (!ListingsDataRunner.finishedSeeding) {
+            LOGGER.info(
+                "Database not seeded yet. Skipping scheduled update of ListingDailyInfoPrices."
+            );
+            return;
+        }
+
+        LOGGER.info("Starting scheduleListingInfoUpdates");
+
+        ZoneId zoneBelgrade = ZoneId.of("Europe/Belgrade");
+        ZonedDateTime nowBelgrade = ZonedDateTime.now(zoneBelgrade);
+        // ZonedDateTime nowBelgrade = ZonedDateTime.now(zoneBelgrade) .plusDays(1); // only for
+        // testing
+        ZonedDateTime startOfToday = nowBelgrade.truncatedTo(ChronoUnit.DAYS);
+
+        OffsetDateTime startOfYesterday =
+            startOfToday.minusDays(1)
+                .toOffsetDateTime();
+        OffsetDateTime endOfYesterday = startOfToday.toOffsetDateTime();
+
+        LOGGER.info("startOfYesterday " + startOfYesterday + " endOfYesterday " + endOfYesterday);
+
+        List<ListingDailyPriceInfo> ldpis = new ArrayList<>();
+        List<Security> securities = securityRepository.findAll();
+        for (Security s : securities) {
+
+            ListingDailyPriceInfo ldpi =
+                makeYesterdaysListingDailyPriceInfo(s, startOfYesterday, endOfYesterday);
+            if (ldpi == null) continue;
+
+            Optional<List<ListingDailyPriceInfo>> yesterdaysInfoOptional =
+                listingDailyPriceInfoRepository.getListingDailyPriceInfoForDate(
+                    s.getId(),
+                    startOfYesterday,
+                    endOfYesterday
+                );
+
+            if (
+                yesterdaysInfoOptional.isPresent()
+                    && !yesterdaysInfoOptional.get()
+                        .isEmpty()
+            ) {
+                LOGGER.error(
+                    "There is already yesterdays listing info! Count: "
+                        + yesterdaysInfoOptional.get()
+                            .size()
+                        + ". Security: "
+                        + s.getTicker()
```

Comment:
never concatenate for logging, use a format string. https://slf4j.org/api/org/slf4j/helpers/MessageFormatter.html